ckanext/security/resource_upload_validator.py

```diff
@@ -0,0 +1,98 @@
+import mimetypes
+import magic
+import logging
+import os
+
+from ckan.logic import ValidationError
+from ckan.common import config
+
+log = logging.getLogger(__name__)
+
+DEFAULT_UPLOAD_BLACKLIST = ['.exe']
+DEFAULT_EXTENDED_UPLOAD_MIMETYPES = { 'application/x-dosexec': '.exe' }
+
+def _add_mimetypes():
+    if not mimetypes.inited:
+        mimetypes.init()
+
+    # Add mimetypes from config
+    config_mimetypes = eval(config.get('ckanext.security.extended_upload_mimetypes', '{}'))
```

Comment:
See README for explanation of this config, but it basically allows improved matching between the mime types detected by `python-magic` and the mappings from mime type to extension in `mimetypes`